containers-tests/tests/intmap-properties.hs

```diff
@@ -1187,3 +1525,20 @@ prop_fromSet :: [(Int, Int)] -> Bool
 prop_fromSet ys =
   let xs = List.nubBy ((==) `on` fst) ys
   in fromSet (\k -> fromJust $ List.lookup k xs) (IntSet.fromList $ List.map fst xs) == fromList xs
+
+newtype Identity a = Identity a
+
+instance Functor Identity where
+  fmap f (Identity a) = Identity (f a)
+
+instance Applicative Identity where
+  pure a = Identity a
+  Identity f <*> Identity a = Identity (f a)
+
+prop_traverseWithKey :: IntMap () -> Bool
+prop_traverseWithKey mp = mp == newMap
+  where Identity newMap = traverseWithKey (\_ -> Identity) mp
+
+prop_traverseMaybeWithKey :: IntMap () -> Bool
```

Comment:
We want to generalize this in two directions:

1. Show that it degrades to `traverseWithKey` when used with `fmap Just`.
2. Show that it degrades to `mapMaybeWithKey` when used with `Identity`.